src/Simplic.OxS.ModelDefinition.Service/ModelDefinitionService.cs

```diff
@@ -0,0 +1,365 @@
+
+            modelDefinition.Operations.Create = new OperationDefinition
+            {
+                Endpoint = postAttribute.Endpoint,
+                RequestReference = GetReferenceName(postAttribute.Request),
+                ResponseReference = GetReferenceName(postAttribute.Response),
+                Type = "http-post"
+            };
+
+            BuildPropertiesOrReference(modelDefinition, postAttribute.Request);
+            BuildPropertiesOrReference(modelDefinition, postAttribute.Response);
+        }
+
+        private static void BuildPatch(ModelDefinition modelDefinition, MethodInfo patchMethod)
+        {
+
+            var attribute = Attribute.GetCustomAttribute(patchMethod, typeof(ModelDefinitionPatchOperationAttribute));
+
+            if (attribute is not ModelDefinitionPatchOperationAttribute patchAttribute)
+                return;
+
+            modelDefinition.Operations.Update = new OperationDefinition
+            {
+                Endpoint = patchAttribute.Endpoint,
+                RequestReference = GetReferenceName(patchAttribute.Request),
+                ResponseReference = GetReferenceName(patchAttribute.Response),
+                Type = "http-patch"
+            };
+
+            BuildPropertiesOrReference(modelDefinition, patchAttribute.Request);
+            BuildPropertiesOrReference(modelDefinition, patchAttribute.Response);
+        }
+
+        private static void BuildDelete(ModelDefinition modelDefinition, MethodInfo deleteMethod)
+        {
+            var attribute = Attribute.GetCustomAttribute(deleteMethod, typeof(ModelDefinitionDeleteOperationAttribute));
+
+            if (attribute is not ModelDefinitionDeleteOperationAttribute deleteAttribute)
+                return;
+
+            modelDefinition.Operations.Delete = new OperationDefinition
+            {
+                Endpoint = deleteAttribute.Endpoint,
+                Type = "http-delete"
+            };
+        }
+
+        private static void BuildPropertiesOrReference(ModelDefinition modelDefinition, Type response)
+        {
+            if (modelDefinition.Model != null
+                &&
+                modelDefinition.Model.Equals(GetReferenceName(response)))
+                // Return since the type is alrady the model type.
+                return;
+
+            if (modelDefinition.Model == null)
+            {
+                modelDefinition.Model = GetReferenceName(response);
+                modelDefinition.Properties = BuildProperties(response, modelDefinition);
+            }
+            else
+            {
+                AddRefereceDefinition(modelDefinition, response);
+            }
+        }
+
+        #endregion
+
+        #region Properties
+
+        private static IList<PropertyDefinition> BuildProperties(Type model, ModelDefinition modelDefinition)
+        {
+            var properties = new List<PropertyDefinition>();
+            foreach (var property in model.GetProperties())
```

Comment:
*Important*

Here you are reading all properties from the model. Does this support inheritance. For example:

```cs
public class ModelA 
{
    public string Name { get; set; }
}

public class ModelB : ModelA
{
    public string Street { get; set; }
}

```

if properties of `ModelB` are read?

Reply:
Yes, in the given example e.g the create tour request contains just a collection of actions itself, other properties come from an inherited class and all of the properties are  present in the model definition.

Reply:
Perfect